core/server/master/src/test/java/alluxio/master/block/BlockMasterTest.java

```diff
@@ -106,10 +116,13 @@ public void before() throws Exception {
     mClock = new ManualClock();
     mExecutorService =
         Executors.newFixedThreadPool(2, ThreadFactoryUtils.build("TestBlockMaster-%d", true));
-    mBlockMaster = new DefaultBlockMaster(mMetricsMaster, masterContext, mClock,
-        ExecutorServiceFactories.constantExecutorServiceFactory(mExecutorService));
+    mBlockMaster = spy(new DefaultBlockMaster(mMetricsMaster, masterContext, mClock,
```

Comment:
You can write an easy extension of `DefaultBlockMaster` to override `getClusterId()` logic. We prefer that to using mocks because it's much easier to maintain. For example if the method signature changes, mocks will fail (change behavior) silently but children classes will complain.